File: Examples/CSharp/Knowledge-Base/SaveDiagramTo-VDX-PDF-JPEG-withAspose.cs
using Aspose.Diagram;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CSharp.Knowledge_Base
{
    public class SaveDiagramTo_VDX_PDF_JPEG_withAspose
    {
        public static void Run() 
        {
            // ExStart:SaveDiagramTo_VDX_PDF_JPEG_withAspose
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_KnowledgeBase();

            // load an exiting Visio diagram
            Diagram vsdDiagram = new Diagram(dataDir + "Drawing1.vsd");

            // Save the diagram as VDX
            vsdDiagram.Save(dataDir + "SaveDiagramToVDXwithAspose_Out.vdx", SaveFileFormat.VDX);

            // Save as PDF
            vsdDiagram.Save(dataDir + "SaveDiagramToPDFwithAspose_Out.pdf", SaveFileFormat.PDF);

            // Save as JPEG
            vsdDiagram.Save(dataDir + "SaveDiagramToJPGwithAspose_Out.jpg", SaveFileFormat.JPEG);
            // ExEnd:SaveDiagramTo_VDX_PDF_JPEG_withAspose
        }
    }
}

File: Examples/VisualBasic/Knowledge-Base/SaveDiagramTo-VDX-PDF-JPEG-withAspose.vb
Imports Aspose.Diagram
Imports System
Public Class SaveDiagramTo_VDX_PDF_JPEG_withAspose
    Public Shared Sub Run()
        ' ExStart:SaveDiagramTo_VDX_PDF_JPEG_withAspose
        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_KnowledgeBase()

        ' Load an exiting Visio diagram
        Dim vsdDiagram As New Diagram(dataDir & Convert.ToString("Drawing1.vsd"))

        ' Save the diagram as VDX
        vsdDiagram.Save(dataDir & Convert.ToString("SaveDiagramToVDXwithAspose_out.vdx"), SaveFileFormat.VDX)

        ' Save as PDF
        vsdDiagram.Save(dataDir & Convert.ToString("SaveDiagramToPDFwithAspose_out.pdf"), SaveFileFormat.PDF)

        ' Save as JPEG
        vsdDiagram.Save(dataDir & Convert.ToString("SaveDiagramToJPGwithAspose_out.jpg"), SaveFileFormat.JPEG)
        ' ExEnd:SaveDiagramTo_VDX_PDF_JPEG_withAspose
    End Sub
End Class
